app/assets/javascripts/views/edit.js

```diff
@@ -0,0 +1,18 @@
+var editDocument = document.querySelectorAll('.edit_document')
+if (editDocument) {
+  document.querySelectorAll('[data-warn-before-unload=true]').forEach(function (element) {
```

Comment:
Rather than having this set on a particular form input could we have this set on a form itself.

So we can have `<form data-warn-before-unload=true>`  and just listen for change events that have bubbled up from it? 

Then we can just listen to the submit event on the form too rather than having to put an event specifically on buttons within the form.

Reply:
@kevindew - thanks for this, this was a good idea. This has been done.

Reply:
Great thanks 👍